org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/util/IResourcesSetupUtil.java

```diff
@@ -370,7 +370,7 @@ public static void waitForBuild(IProgressMonitor monitor) {
 		try {
 			ResourcesPlugin.getWorkspace().build(IncrementalProjectBuilder.INCREMENTAL_BUILD, monitor);
 		} catch (CoreException e) {
-			throw new OperationCanceledException(e.getMessage());
+			throw new RuntimeException(e.getMessage(), e);
```

Comment:
can you also fix this in 
org.eclipse.xtext.junit4.ui.util.IResourcesSetupUtil.waitForBuild(IProgressMonitor)

i have no idea why there is OperationCanceledException in the first place. maybe @szarnekow remembers

Reply:
Sadness: in `#setAutobuild` above we use sneakThrow. 
Though I prefer some consistency, I don't see a good reason why this should be OperationCanceledException except for the fact that some of the APIs in Xtext will gracefully rethrow OCE but might catch other plain RuntimeException.

OperationCanceledException#addSuppressed(e) is probably not much different from the new runtime exception. What do you think?

Reply:
@szarnekow do you mean something like this?

```java
OperationCanceledException operationCanceledException = new OperationCanceledException(e.getMessage());
operationCanceledException.addSuppressed(e);
throw operationCanceledException;
```

if the result is the same (the stacktrace of the original exception is shown) I'd be fine

Reply:
Yes, it'd be printed with the stacktrace.

Reply:
@szarnekow OK, I've pushed-forced with your suggestion
@cdietrich I've also updated the same method in org.eclipse.xtext.junit4.ui.util.IResourcesSetupUtil